builder/actions/setup_ci_from_files.py

```diff
@@ -0,0 +1,301 @@
+                        environment_multi[environment_name + "_SECRET_ACCESS_KEY"] = arn_credentials[1]
+                        environment_multi[environment_name + "_SESSION_TOKEN"] = arn_credentials[2]
+                    except Exception as ex:
+                        sys.exit(
+                            f"[FAIL] {environment_name} [Input Role ARN]: Exception ocurred trying to get role ARN credentials")
+
+                # The same as "input_role_arn" but instead of taking the IAM role arn directly, it instead
+                # takes a AWS Secret Name and assumes the value in that secret is the IAM Role ARN to use and assume.
+                # Valid JSON:
+                #    'input_role_arn_secret': <AWS Secret Name containing IAM Role ARN here>
+                if ('input_role_arn_secret' in item):
+                    try:
+                        input_role_arn = self.env_instance.shell.get_secret(str(item['input_role_arn_secret']))
+                        arn_credentials = self.get_arn_role_credentials(input_role_arn)
+                        environment_value = "SUCCESS"
+                        environment_multi[environment_name + "_ACCESS_KEY"] = arn_credentials[0]
+                        environment_multi[environment_name + "_SECRET_ACCESS_KEY"] = arn_credentials[1]
+                        environment_multi[environment_name + "_SESSION_TOKEN"] = arn_credentials[2]
+                    except:
+                        sys.exit(
+                            f"[FAIL] {environment_name} [Input Role ARN Secret]: Exception ocurred trying to get role ARN credentials")
+
+                # Downloads the S3 file at the given URL and sets environment_value to the downloaded (temporary) file.
+                # Valid JSON:
+                #   'input_s3': <S3 URL Here>
+                if ('input_s3' in item):
+                    try:
+                        tmp_file = tempfile.NamedTemporaryFile()
+                        tmp_file.flush()
+                        self.tmp_file_storage.append(tmp_file)
+                        tmp_s3_filepath = tmp_file.name
+                        self.copy_s3_file(str(item['input_s3']), tmp_s3_filepath)
+                        environment_value = str(tmp_s3_filepath)
+                    except Exception as ex:
+                        sys.exit(f"[FAIL] {environment_name} [Input S3]: Exception ocurred trying to get S3 file")
+
+                ############################################################
+                # OS Condition (starts with "os_")
+                ############################################################
+
+                # Checks to see if the operating system is the same as the desired operating system(s) of the environment variable.
+                # Valid JSON:
+                #   'os_only': "windows" or "linux" or "mac" <add ',' for multiple. Example: "windows,linux">
+                if ('os_only' in item):
```

Comment:
we already have concept of host/target configs in builder, why are we introducing another similar concept inside env config files?

Reply:
The idea is that we use the exact same file for all of the CRTs and SDKs and for all platforms, therefore we need a way to differentiate between them when setting environment variables that are OS specific (like Windows certificate store, for example). I was initially going to do something like `+config` and `!config` so we could just add/remove/override the JSON/XML files, but as discussed in the internal thread on Slack, this would be complicated due to how the override system works. I thought it was just better to add the config check directly to the environment variables file.

Reply:
we support adding custom stuff pretest (initializing env variables, etc...), are there limitations with that, that make it not usable for env variable override? 
my concern with adding a whole new format for specifying those env variables is that its adding a lot of complexity to an already overly complex way to build thing

Reply:
We could, but then we'd need a special builder action for each of the CI tests that have OS specific requirements. Granted sometimes this cannot be avoided, like with PKCS11 or Windows Cert Connect due to needing additional setup.

But on a whole, I think the idea is to have a single file that is the main set of environment variables so adding/removing/modifying environment variables is a one-stop-shop solution. Moving the specific environment variable setups into a bunch of builder actions would spread the setup of the environment variables across a bunch of different files.

That is not to say adding a way to change how the environment variable based on OS in the JSON/XML is the best solution, but I couldn't find a decent solution that still keeps the environment files all together and not duplicating builder action scripts across CRT and SDK repos.

I don't know... Will need to think further on this...